ftl/code/controllers/subsystem/ftl_navigation.dm

```diff
@@ -0,0 +1,220 @@
+	for(var/datum/sector/S in oldS.connected_sectors) //Cleans up unvisited sectors.
+		oldS.connected_sectors -= S
+		if(S != toS) qdel(S)
+
+
+/datum/controller/subsystem/ftl_navigation/proc/generate_connecting_sectors(var/datum/sector/S)
+	var/datum/sector/N
+	var/new_sector
+	for(var/a in 1 to rand(1,3))
+		new_sector = pickweight(sector_types)
+		N = new new_sector
+		if(N.unique) sector_types -= new_sector //If the sector is unique remove it from the spawn pool
+		S.connected_sectors += N
+	S.generated_sectors = TRUE
+
+
+
+/datum/controller/subsystem/ftl_navigation/proc/ftl_transition(var/ftl_start = TRUE)
+	message_admins("Changing FTL effects")
+	ftl_parallax(ftl_start)
+	ftl_handle_transit_turfs(ftl_start)
+	if(ftl_start)
+		ftl_state = FTL_JUMPING
+		message_admins("FTL has begun")
+		addtimer(CALLBACK(src, .proc/ftl_transition, FALSE), 100)
+	else leave_ftl()
+
+
+/datum/controller/subsystem/ftl_navigation/proc/leave_ftl()
+	current_sector = to_sector
+	current_sector.visited = TRUE
+	visited_sectors += to_sector
+	to_sector = null
+	for(var/datum/sector/S in current_sector.connected_sectors)
+		generate_connecting_sectors(S)
+	ftl_state = FTL_IDLE
+	message_admins("FTL has ended")
+
+
+/datum/controller/subsystem/ftl_navigation/proc/ftl_parallax(var/ftl_start = TRUE)
+	for(var/area/A in ship_areas)
+		A.parallax_movedir = ftl_start ? 4 : 0
+		for(var/atom/movable/AM in A)
+			if(length(AM.client_mobs_in_contents))
+				AM.update_parallax_contents()
+
+
+/datum/controller/subsystem/ftl_navigation/proc/ftl_handle_transit_turfs(var/ftl_start = TRUE)
+	var/throw_dir
+	var/flavor_text
+	if(ftl_start)
+		throw_dir = WEST
+		flavor_text = "<span class='notice'>You feel the ship lurch as it enters FTL.</span>"
+	else
+		throw_dir = EAST
+		flavor_text = "<span class='notice'>You feel the ship lurch as it exits FTL.</span>"
+	for(var/z in SSmapping.levels_by_trait(ZTRAIT_STATION))
+		for(var/datum/sub_turf_block/STB in split_block(locate(1,1,z), locate(255,255,z)))
+			for(var/turf/T in STB.return_list())
+				if(istype(T,/turf/open/space/transit/ftl))
+					var/turf/open/space/transit/ftl/F = T
+					F.begin_ftl()
+				for(var/obj/machinery/light/L in T.contents) //Makes lights flicker, since I liked how lighting would falter during FTL
+					L.flicker (4,15)
+				for(var/mob/living/M in T.contents) //Messing with players
+					if(M.buckled && M.client) shake_camera(M,3,1)
+					else
+						if(M.client)
+							shake_camera(M,8,1)
+							to_chat(M,flavor_text)
+						if(jump_throw_force)
+							var/turf/target = get_edge_target_turf(M, throw_dir)
+							var/range = jump_throw_force
+							var/speed = range/5
+							M.throw_at(target,range,speed)
+						if(jump_knockdown_force)
+							M.Knockdown(jump_knockdown_force)
+
+
+// /datum/controller/subsystem/starmap/proc/ftl_message(var/message)
+// 	for(var/obj/machinery/computer/ftl_navigation/C in ftl_consoles)
+// 		C.status_update(message)

```

Comment:
If unused or not planning to use you should remove this

Reply:
I plan to use this later

Reply:
ok cool